DS-400 A components usage guidelines (#4504)
coveo/react-vapor: packages/storybook/src/call-to-action/ActionIcon-usage.mdx

Reviewer: I would rename these files so they match the name of the components it is referring to

```
packages/storybook/src/call-to-action/ActionIcon.mdx
```

import {Meta} from '@storybook/addon-docs/blocks';
import {Table} from '@mantine/core';

> <Meta title="@components/call-to-action/ActionIcon/Usage" />

Reviewer: We should add the `of` prop here so that it properly ties the docs page to the component.

```suggestion
import * as stories from './ActionIcon.stories';

<Meta of={stories} title="@components/call-to-action/ActionIcon" />
```

|Before|After|
|---|---|
|<img width="288" height="122" alt="Image" src="https://github.com/user-attachments/assets/f85fa15c-bd0a-4ffb-a017-527fb1bdf984" />|<img width="289" height="123" alt="Image" src="https://github.com/user-attachments/assets/fd66ad73-dccf-463c-a277-e704d5773a26" />|

{/* storybook-usage-guidelines: rewritten */}

# ActionIcon

Icon-only button that can trigger actions and can show a disabled-state tooltip.

## Overview

Use ActionIcon when an action is best represented by an icon alone — for example, delete, close, or edit controls in compact layouts where a full button label would clutter the UI.

It extends Mantine's ActionIcon with support for async click handlers and a tooltip that appears when the button is disabled, letting you explain why the action isn't available.

> ## Props

Reviewer: I would delete the following sections from the `Docs` page because this information is pretty much just a duplication of what devs will find in the `Demo` page of the component:
- Props
- Sub-Components
- Usage

Basically all that is the API reference part that would be meant for developers. We want developers to go look at the `Demo` directly, play with the controls panel and look at the `Code` tab.

        </Table.Tr>
        <Table.Tr>
            <Table.Td>`ActionIcon.DestructiveSecondary`</Table.Td>
            <Table.Td>High emphasis for destructive actions</Table.Td>
        </Table.Tr>
        <Table.Tr>
            <Table.Td>`ActionIcon.DestructiveTertiary`</Table.Td>
            <Table.Td>Medium emphasis for destructive actions</Table.Td>
        </Table.Tr>
        <Table.Tr>
            <Table.Td>`ActionIcon.DestructiveQuaternary`</Table.Td>
            <Table.Td>Low emphasis for destructive actions</Table.Td>
        </Table.Tr>
    </Table.Tbody>
</Table>

## Usage

```tsx
import {ActionIcon} from '@coveord/plasma-mantine';
import {IconTrash} from '@coveord/plasma-react-icons';

const removeItem = () => {};
const deleteItem = async () => {};

// Common pattern: use a sub-component for the intended emphasis.
<ActionIcon.Quaternary onClick={removeItem}>
    <IconTrash aria-label="Remove" />
</ActionIcon.Quaternary>;

// Async handlers automatically show a loading state while pending.
<ActionIcon.DestructivePrimary onClick={deleteItem}>
    <IconTrash aria-label="Delete" />
</ActionIcon.DestructivePrimary>;

// Disabled actions MAY explain why they are unavailable.

> <ActionIcon.Primary disabled disabledTooltip="You do not have permission to remove this item">

Author: ```suggestion
<ActionIcon.Primary disabled disabledTooltip="You don't have permission to remove this item.">
```
